### PR TITLE
[New] Add tracing.js w/ git auto-update via single package.json

### PR DESCRIPTION
**Pull request by tulikavijay**
Pull request for issue: #11973
Related issue(s): # #

Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required): https://github.com/ebobby/tracing.js
 * Official website (optional, not the repository):
 * NPM package url (optional):
 * License and its reference: MIT
 * GitHub / Bitbucket popularity (required):
   - Count of watchers: 24
   - Count of stars: 516
   - Count of forks: 18
 * NPM download stats (optional):
   - Downloads in the last day:
   - Downloads in the last week:
   - Downloads in the last month:

# Essential checklist
 * [ ] I'm the author of this library
   * [ ] I would like to add link to the page of this library on CDNJS on website / readme
 * [x] This lib was not found on cdnjs repo
 * [ ] No already exist / duplicated issue and PR
* [x] The lib has notable popularity
   * [x] More than 200 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [ ] More than 800 downloads stats per month on npm registry
 * [ ] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [ ] Has valid tags for each versions (for git auto-update)
 * [x] Auto-update setup
 * [x] Auto-update target/source is valid.
 * [x] Auto-update filemap is correct.

# Git commit checklist
 * [x] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [x] The parent of the commit(s) in the PR is not old than 3 days.
 * [x] Pull request is sending from a non-master branch with meaningful name.
 * [ ] Separate unrelated changes into different commits.
 * [ ] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [ ] Close corresponding issue in commit message
 * [x] Mention related issue(s), people in commit message, comment.

### CODE REVIEW

File: ajax/libs/tracing.js/package.json [hunk 1]
```diff
@@ -0,0 +1,34 @@
+{
+	"name": "tracing-js",
+	"filename": "tracing.min.js",
+	"description": "Tracing.js is a library/utility to help you debug and trace javascript code. Tracing.js allows you to set functions of your own to be called before or after target functions in case of debugging get called. It is like events for function invocations.",
```
Comment (extend1994) [inline ajax/libs/tracing.js/package.json]: Let's simplify description. Only use `Tracing.js is a library/utility to help you debug and trace javascript code.`

File: ajax/libs/tracing.js/package.json [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+	"license": "MIT",
+	"author": "ebobby",
```
Comment (extend1994) [inline ajax/libs/tracing.js/package.json]: Not sure should we use ID or author's name here? cc @cdnjs/intern3 @cdnjs/library-reviewer

File: ajax/libs/tracing.js/package.json [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+		"type": "git",
+		"url": "https://github.com/ebobby/tracing.js"
+	},
+	"keywords": [
+		"tracing",
+		"debug",
+		"development",
+		"javascript"
+	],
+	"npmName": "tracing-js",
```
Comment (extend1994) [inline ajax/libs/tracing.js/package.json]: This library doesn't have npm package. Please remove L17-L23.

File: ajax/libs/tracing.js/package.json [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+		"basePath": "dist",
+		"files": [
+			"**/!(*.gz)"
+		]
+	}],
+	"autoupdate": {
+		"source": "git",
+		"target": "git://github.com/ebobby/tracing.js",
+		"fileMap": [{
+			"basePath": "dist",
```
Comment (extend1994) [inline ajax/libs/tracing.js/package.json]: Let's use `""` here. The file we need is under root directory.

File: ajax/libs/tracing.js/package.json [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+			"files": [
+				"**/!(*.gz)"
```
Comment (extend1994) [inline ajax/libs/tracing.js/package.json]: Let's use `tracing*` to fetch `tracing.js`. For more details about this pattern, please see https://github.com/isaacs/node-glob

File: ajax/libs/tracing.js/package.json [hunk 6]
```diff
@@ -0,0 +1,34 @@
+{
+	"name": "tracing-js",
```
Comment (extend1994) [inline ajax/libs/tracing.js/package.json]: `name` should be the same as your directory name and repo name, which would be `tracing.js`.

**Comment by tulikavijay** [comment]
Thankyou for being patient and reviewing the PR. I'll make changes shortly.

File: ajax/libs/tracing.js/package.json [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+    },
+  "keywords": [
+    "tracing",
+    "debug",
+    "development",
+    "javascript"
+    ],
+  "autoupdate": {
+    "source": "git",
+    "target": "git://github.com/ebobby/tracing.js",
```
Comment (PeterDaveHello) [inline ajax/libs/tracing.js/package.json]: Can you please specify the trailing `.git` in the url? Thanks.

**Pull request merged**